crates/api/src/peer_meta_store.rs

```diff
@@ -25,16 +31,46 @@ pub trait PeerMetaStore: 'static + Send + Sync + std::fmt::Debug {
 
     /// Mark a peer url unresponsive with an expiration timestamp.
     ///
+    /// The value that will be stored with the peer key is the passed in timestamp
+    /// from when the URL became unresponsive.
+    ///
     /// After the expiry timestamp has passed, the peer url is supposed to be removed
     /// from the store.
     fn mark_peer_unresponsive(
         &self,
         peer: Url,
         expiry: Timestamp,
-    ) -> BoxFuture<'_, K2Result<()>>;
+        when: Timestamp,
+    ) -> BoxFuture<'_, K2Result<()>> {
+        Box::pin(async move {
+            self.put(
+                peer.clone(),
+                format!("{KEY_PREFIX_ROOT}:{META_KEY_UNRESPONSIVE}"),
+                serde_json::to_vec(&when).map_err(K2Error::other)?.into(),
+                Some(expiry),
+            )
+            .await?;
+            Ok(())
+        })
+    }
 
-    /// Check if a peer is marked unresponsive in the store.
-    fn is_peer_unresponsive(&self, peer: Url) -> BoxFuture<'_, K2Result<bool>>;
+    /// Get the timestamp of when a peer URL was marked unresponsive, if it is present in the
+    /// store.
+    fn get_unresponsive_url(
```

Comment:
```suggestion
    fn get_unresponsive_for_url(
```

Reply:
I would actually still find `is_peer_unresponsive()` appropriate as a name.

Reply:
How about this?

Reply:
I renamed to `get_unresponsive` and `mark_unresponsive`, but then though it's more consistent to match the get/set pattern and ended up with `get_unresponsive` and `set_unresponsive`.